Sample Applications/WPFGallery/Navigation/NavigationService.cs

```diff
@@ -40,12 +43,59 @@ public class NavigationService : INavigationService
 
     public event EventHandler<NavigatingEventArgs> Navigating;
 
+    private Dictionary<string, Type> pageNameToTypeMapping;
+
 
     public NavigationService(IServiceProvider serviceProvider)
     {
         _serviceProvider = serviceProvider;
         _history = new Stack<Type>();
         _future = new Stack<Type>();
+
+        initializeMapping();
+    }
+
+    private void initializeMapping()
```

Comment:
Don't do it this way. There is a ControlsInfoData.json, which is to be used to get the list of all the controls and pages. Use that.

This will reintroduce the redundancy of data.